src/ansiblelint/rules/name.py

```diff
@@ -11,20 +11,47 @@
 if TYPE_CHECKING:
     from typing import Optional
 
+    from ansiblelint.constants import odict
     from ansiblelint.file_utils import Lintable
 
 
 class NameRule(AnsibleLintRule):
-    """Rule for checking task names and their content."""
+    """Rule for checking task and play names."""
 
     id = "name"
     description = (
-        "All tasks should have a distinct name for readability "
+        "All tasks and plays should have a distinct name for readability "
         "and for ``--start-at-task`` to work"
     )
     severity = "MEDIUM"
     tags = ["idiom"]
-    version_added = "historic"
+    version_added = "v6.5.0 (last update)"
+
+    def matchplay(
+        self, file: "Lintable", data: "odict[str, Any]"
+    ) -> List["MatchError"]:
+        """Return matches found for a specific play (entry in playbook)."""
+        if file.kind != "playbook":
+            return []
+        if "name" not in data and not any(
+            key in data
+            for key in ["import_playbook", "ansible.builtin.import_playbook"]
+        ):
+            return [
+                self.create_matcherror(
+                    message="All plays should be named.",
+                    linenumber=data[LINE_NUMBER_KEY],
+                    tag="name[play]",
+                    filename=file,
+                )
+            ]
+        # pylint: disable=assignment-from-none
```

Comment:
Does pylint still complain if you remove this now that you fixed the return of _check_name?